pkg/api/api.go

```diff
@@ -154,16 +159,17 @@ func (hs *HTTPServer) registerRoutes() {
 
 		// users (admin permission required)
 		apiRoute.Group("/users", func(usersRoute routing.RouteRegister) {
-			usersRoute.Get("/", routing.Wrap(SearchUsers))
-			usersRoute.Get("/search", routing.Wrap(SearchUsersWithPaging))
-			usersRoute.Get("/:id", routing.Wrap(GetUserByID))
-			usersRoute.Get("/:id/teams", routing.Wrap(GetUserTeams))
-			usersRoute.Get("/:id/orgs", routing.Wrap(GetUserOrgList))
+			const userIDScope = `users:{{ index . ":id" }}`
+			usersRoute.Get("/", authorize(reqGrafanaAdmin, accesscontrol.ActionUsersRead, accesscontrol.ScopeUsersAll), routing.Wrap(SearchUsers))
+			usersRoute.Get("/search", authorize(reqGrafanaAdmin, accesscontrol.ActionUsersRead, accesscontrol.ScopeUsersAll), routing.Wrap(SearchUsersWithPaging))
+			usersRoute.Get("/:id", authorize(reqGrafanaAdmin, accesscontrol.ActionUsersRead, userIDScope), routing.Wrap(GetUserByID))
```

Comment:
Is this a change in access level?

No, for two reasons:
- These endpoints do not call something that isn't in the background called by the `/user` endpoint accessible to users.
- We don't actually grant users access to `users:read` for `users:<id>` but only for the literal `users:self`. This might change, I think it'd be reasonable to change that, but that means that _at the moment_ this is not a change in scope.